### ISSUE TITLE
Change Request: Allow the same rule to appear several times under different names (rule synonyms)

### ISSUE THREAD

**Issue by ilyub** [eslint/eslint#15638]
### ESLint version

8.9.0

### What problem do you want to solve?

Consider "no-restricted-syntax" rule that forbids 20 restricted syntaxes: syntax1, syntax2, etc:
```ts
"no-restricted-syntax": [
  "error",
  {
    "selector": "selector1",
    "message": "message1"
  },
  {
    "selector": "selector2",
    "message": "message2"
  },
  ...
]
```

Let's say that I want to ignore syntax5 at certain line. Currently I will have to do this:
```ts
// eslint-disable-next-line no-restricted-syntax
<some-code-with-syntax5>
```

The problem is that I am disabling check for all 20 syntaxes, not just for syntax5, which may result in future undiscovered errors.

### What do you think is the correct solution?

Let's introduce rule synonyms:
```ts
"no-restricted-syntax/syntax1": [
  "error",
  {
    "selector": "selector1",
    "message": "message1"
  },
],
"no-restricted-syntax/syntax2": [
  {
    "selector": "selector2",
    "message": "message2"
  }
],
...
```

Now I have neat way to disable only what I want:
```ts
// eslint-disable-next-line no-restricted-syntax/syntax5
<some-code-with-syntax5>
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
Duplicate of #15620
